src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -51,32 +54,104 @@
 	private static final String CANVAS_API_ADD_USER = "canvas.api.add.user.regex";
 	private static final String CANVAS_API_GET_COURSE = "canvas.api.get.single.course.regex";
+	private static final String MANAGER_SERVLET_NAME = "/manager";
 
 	private static final String DELETE = "DELETE";
 	private static final String POST = "POST";
 	private static final String GET = "GET";
 	private static final String PUT = "PUT";
+	
 	private String canvasToken;
 	private String canvasURL;
-	ResourceBundle props = ResourceBundle.getBundle("sectiontool");
-
+	private String callType;
+	private String ltiUrl;
+	private String ltiKey;
+	private String ltiSecret;
 	
+	private final static String CCM_PROPERTY_FILE_PATH = "ccmPropsPath";
+	private final static String CCM_SECURE_PROPERTY_FILE_PATH = "ccmPropsPathSecure";	
+	
+	protected static Properties appExtSecurePropertiesFile=null;
+	protected static Properties appExtPropertiesFile=null;	
+
 	public void init() throws ServletException {
 		M_log.debug(" Servlet init(): Called");
+		appExtPropertiesFile = Utils.loadProperties(CCM_PROPERTY_FILE_PATH);
+		appExtSecurePropertiesFile = Utils.loadProperties(CCM_SECURE_PROPERTY_FILE_PATH);		
 	}
 	
-	protected void doGet(HttpServletRequest request,HttpServletResponse response){
+	public void fillContext(Context context, HttpServletRequest request) {
+		M_log.debug("fillContext() called");		
+		ViewToolContext vtc = (ViewToolContext)context;
+//		test code
+//		vtc.put("variable", "happiness");
+	}
+
+	// Deal nicely with error conditions.
+	public void doError(HttpServletRequest request, HttpServletResponse response, String s)
+			throws java.io.IOException {
+
+		StringBuilder return_url = new StringBuilder();
+		if (return_url != null && return_url.length() > 1) {
+			return_url.append((return_url.indexOf("?") > 1) ? "&" : "?");
+			return_url.append("lti_msg=").append(URLEncoder.encode(s,"UTF-8"));
+			response.sendRedirect(return_url.toString());
+			return;
+		}
+		PrintWriter out = response.getWriter();
+		out.println(s);
+	}	
+	
+	public void doGet(HttpServletRequest request,HttpServletResponse response){
 		M_log.debug("doGet: Called");
 		try {
-			canvasRestApiCall(request, response);
+			if(request.getServletPath().equals(MANAGER_SERVLET_NAME)){
+				
+				canvasRestApiCall(request, response);
+			}
+			else{
+				M_log.info("NOT MANAGER");
+				doRequest(request, response); //doRequest will always call fillContext()
+			}
 		}catch(Exception e) {
 			M_log.error("GET request has some exceptions",e);
 		}
 	}
 	
-	protected void doPost(HttpServletRequest request,HttpServletResponse response){
+	public void doPost(HttpServletRequest request,HttpServletResponse response){
 		M_log.debug("doPOST: Called");
 		try {
+			//determine if this is an LTI Call or a browser call?
+			if(request.getParameterMap().containsKey("oauth_consumer_key")){
+				for (Object e : request.getParameterMap().entrySet()) {
```

Comment:
Extract out contents of the "if true" clause into a separate method for clarity.  To deal with the nested return have the method return a boolean to tell the caller if it should return.